src/actions/get_bucket_policy.rs

```diff
@@ -0,0 +1,128 @@
+use std::iter;
+use std::time::Duration;
+
+use serde::Deserialize;
+use time::OffsetDateTime;
+use url::Url;
+
+use super::S3Action;
+use crate::actions::Method;
+use crate::signing::sign;
+use crate::sorting_iter::SortingIterator;
+use crate::{Bucket, Credentials, Map};
+
+const POLICY_PARAM: &str = "policy";
+
+/// Retrieve a bucket's policy from S3.
+///
+/// Find out more about `GetBucketPolicy` from the [AWS API Reference][api]
+///
+/// [api]: https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetBucketPolicy.html
+#[derive(Debug, Clone)]
+pub struct GetBucketPolicy<'a> {
+    bucket: &'a Bucket,
+    credentials: Option<&'a Credentials>,
+
+    query: Map<'a>,
+    headers: Map<'a>,
+}
+
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
+pub struct GetBucketPolicyResponse {
+    #[serde(rename = "Version")]
+    pub version: String,
+    #[serde(rename = "Id")]
+    pub id: Option<String>,
+}
+
+impl<'a> GetBucketPolicy<'a> {
+    #[inline]
+    pub fn new(bucket: &'a Bucket, credentials: Option<&'a Credentials>) -> Self {
+        Self {
+            bucket,
+            credentials,
+
+            query: Map::new(),
+            headers: Map::new(),
+        }
+    }
+    pub fn parse_response(s: &str) -> Result<GetBucketPolicyResponse, serde_json::Error> {
```

Comment:
Nit: there should be a separator line here

```suggestion
    }

    pub fn parse_response(s: &str) -> Result<GetBucketPolicyResponse, serde_json::Error> {
```